heidegger_index/validators.py

```diff
@@ -0,0 +1,47 @@
+from django.core.exceptions import ValidationError
+
+import re
+
+## https://wiki.dnb.de/pages/viewpage.action?pageId=48139522
+
+def validate_gnd(value):
+    # ONLY WORKS FOR PND / POST-2012 entries.
+
+    # Validate type
+    if type(value) == str or type(value) == int:
+        # Validate with regex according to https://www.wikidata.org/wiki/Property:P227
+        GND_REGEX = r"1[012]?\d{7}[0-9X]|[47]\d{6}-\d|[1-9]\d{0,7}-[0-9X]|3\d{7}[0-9X]"
+        regex = re.compile(GND_REGEX, re.IGNORECASE)
+        if not regex.match(value):
```

Comment:
Niet nodig om hier die regular expression te compilen, dat doe je meestal als je de pattern vaker wil gebruiken. Anders is het alleen maar extra werk, computationeel gezien. Je kunt gewoon doen:

```python
if not re.match(value, GND_REGEX, re.IGNORECASE)
```

Ook zou ik die `GND_REGEX` even buiten deze functie definiëren als globale variabele.

Reply:
Overigens, deze match gaat volgens mij fout als `value` een `int` is, regular expresson matches werken alleen maar met strings. Dus je moet dat nog even converteren.

Reply:
Hij gooit trouwens wel een fout als ik hem niet compile, dus die laat ik nu staan.